src/app/modules/main/wallet_section/collectibles/module.nim

```diff
@@ -81,49 +81,36 @@ method currentCollectibleModuleDidLoad*(self: Module) =
   self.checkIfModuleDidLoad()
 
 method fetchOwnedCollectibles*(self: Module) =
-  self.controller.fetchOwnedCollectibles(self.chainId, self.address)
+  self.controller.fetchOwnedCollectibles(self.chainId, self.addresses)
 
 method filterChanged*(self: Module, addresses: seq[string], chainIds: seq[int]) =
   let network = self.controller.getNetwork()
-  let account = self.controller.getWalletAccountByAddress(addresses[0])
   self.chainId = network.chainId
-  self.address = account.address
-  self.currentCollectibleModule.setCurrentAddress(network, self.address)
+  self.addresses = addresses
+  self.currentCollectibleModule.setCurrentNetwork(network)
+  self.view.setCollectibles(@[])
+  let data = self.controller.getOwnedCollectibles(self.chainId, self.addresses)
 
-  let data = self.controller.getOwnedCollectibles(self.chainId, self.address)
+  if not data[0].anyLoaded:
+    self.fetchOwnedCollectibles()
 
-  # Trigger a fetch the first time we switch to an account
-  if not data.anyLoaded:
-    self.controller.fetchOwnedCollectibles(self.chainId, self.address)
-
-  self.setCollectibles(self.chainId, self.address, data)
+  self.setCollectibles(data)
 
 proc ownedCollectibleToItem(self: Module, oc: OwnedCollectible): Item =
   let c = self.controller.getCollectible(self.chainId, oc.id)
   let col = self.controller.getCollection(self.chainId, c.collectionSlug)
   return collectibleToItem(c, col, oc.isFromWatchedContract)
 
 method onFetchStarted*(self: Module, chainId: int, address: string) =
-  if self.chainId == chainId and self.address == address:
+  if self.chainId == chainId and address in self.addresses:
     self.view.setIsFetching(true)
 
-method setCollectibles*(self: Module, chainId: int, address: string, data: CollectiblesData) =
-  if self.chainId == chainId and self.address == address:
-    self.view.setIsError(data.isError)
-
-    if data.isError and not data.anyLoaded:
-      # If fetching failed before being able to get any collectibles info,
-      # show loading animation
-      self.view.setIsFetching(true)
-    else:
-      self.view.setIsFetching(data.isFetching)
-
-    var newCollectibles = data.collectibles.map(oc => self.ownedCollectibleToItem(oc))
-    self.view.setCollectibles(newCollectibles)
-    self.view.setAllLoaded(data.allLoaded)
+method resetCollectibles*(self: Module) =
+  let data = self.controller.getOwnedCollectibles(self.chainId, self.addresses)
+  self.setCollectibles(data)
 
 method appendCollectibles*(self: Module, chainId: int, address: string, data: CollectiblesData) =
-  if self.chainId == chainId and self.address == address:
+  if self.chainId == chainId and address in self.addresses:
     self.view.setIsError(data.isError)
 
     if data.isError and not data.anyLoaded:
```

Comment:
hmm we'll need to restructure this a little bit... since this is now called in a loop for each account when the filter changes, it will only consider the state of the last address. You probably want to consider the states of all accounts at the same time here... We should probably ask Ben what we want shown in case one of the accounts selected failed fetching before retrieving any collectibles.

Reply:
Also, in case of no error, setIsFetching to true if any address `isFetching`

Reply:
As it is async, i don't know if that will work. Cause i don't have the state of all account